DirectEmployers/MyJobs
# [PD-1837, PD-1923] Role interface and role API

I addressed too much in previous PRs. This one is solely about:

* Interface for managing roles (PD-1837)
* Associated APIs (PD-1923)

# Most Relevant Files

Thank you @aspidites. I'll try to point this out in the future.

[gulp/gulpfile.js](https://github.com/DirectEmployers/MyJobs/pull/1877/files#diff-ab729026c3d4a4364ad28d329d544093)
[gulp/src/manageusers/manageusers.js](https://github.com/DirectEmployers/MyJobs/pull/1877/files#diff-5b20481ecac9619deecc75649b273934)
[myjobs/views.py](https://github.com/DirectEmployers/MyJobs/pull/1877/files#diff-5419a48adf710fe19aaa02f5ffaf9eee)

# What's New

* Incorporated David's card styles
* CRUD for roles
* Different multiselects (they are React components, not the previous FSM widget)
* Various confirmations to user (e.g. a role must contain an activity, confirm deletion of role, etc.)
* Nice API error messages
* Modularizing more (e.g. cookie.js util by @darrint)

# How to Test

Make sure gulp is installed and:
```
cd /gulp/
npm install
gulp
```
Finally, simply CRUD roles in the browser (JS and more developed Python tests for future PRs): localhost:PORT/manage-users/

# Future Work

* Hide the navigation link to /manage-users/ based on... what criteria?
* Most of the interface for user editing (PD-1836) is done, but that will be in a separate PR.
* All of this is built in Boostrap 3! Kinks will be worked out (tablet/mobile) in future PRs as well as continuing to clean up templates like base-new.html.
* Possibly refactor with Flux (i.e. Actions and the Dispatchers). My code is simple and works, but it can be more elegant and testable.

# You Are Awesome

It's true.
